src/app/shared/forms/new-vehicle-form/new-vehicle-form.component.ts

```diff
@@ -1,12 +1,80 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, ViewEncapsulation, Output, EventEmitter } from '@angular/core';
+import { SelectItem, SelectItemGroup, ConfirmationService, Message, MessageService } from 'primeng/api';
+import { Vehicle } from '@app/shared/models/vehicle';
+import { Manufacturer } from '@app/shared/models/manufacturer';
+import { VehiclesService } from '@app/core/vehicles/vehicles.service';
+
 
 @Component({
   selector: 'app-new-vehicle-form',
   templateUrl: './new-vehicle-form.component.html',
-  styleUrls: ['./new-vehicle-form.component.css']
+  styleUrls: ['./new-vehicle-form.component.css'],
+  encapsulation: ViewEncapsulation.None,
 })
 export class NewVehicleFormComponent implements OnInit {
 
-ngOnInit(){
-}
+  @Output() public vehicleSubmitted: EventEmitter<Vehicle> = new EventEmitter<Vehicle>();
+  private vehicle: Vehicle = new Vehicle();
+  private currentYear: number = new Date().getFullYear();
+  private transmissions: SelectItem[];
+  private manufacturers: Manufacturer[] = [
+    new Manufacturer("Audi", ["A3", "A4", "A5", "A6"]),
+    new Manufacturer("BMW", ["M3", "M5"]),
+    new Manufacturer("Seat", ["Leon", "Toledo"]),
+  ];
+  private filteredManufacturers: any[];
+  private filteredModels: any[];
+
+  constructor(private vehiclesService: VehiclesService, private confirmationService: ConfirmationService, private messageService: MessageService) {
+    this.transmissions = [
+      { label: 'Automatic', value: 'AUTOMATIC' },
+      { label: 'Manual', value: 'MANUAL' }
+    ];
+  }
+
+  ngOnInit() {
+    this.clearForm();
+  }
+
+  clearForm(){
+    this.vehicle = new Vehicle();
+    this.vehicle.productionYear = this.currentYear;
+    this.vehicle.power = 100;
+  }
+
+  filterManufacturers(event) {
+    this.filteredManufacturers = [];
+    for (let i = 0; i < this.manufacturers.length; i++) {
+      let manufacturer = this.manufacturers[i].name;
+      if (manufacturer.toLowerCase().indexOf(event.query.toLowerCase()) == 0) {
```

Comment:
O co tu kaman? wyciągnij to do metody i nazwij

Reply:
Nadal nie wiem jak to nazwać :-1: 

Propozycje? :horse: